test/unit/GraphQLSubscriptions.test.ts

```diff
@@ -38,28 +41,18 @@ describe('GraphQL subscriptions', () => {
   describe('.subscribe', () => {
     it('fetches temporary api key when a new connection is opened', async () => {
       graphqlService.subscribe('subscription { baba }').subscribe(() => {});
-      const spy = jest.fn();
-      graphqlService.WebSocket = spy as any;
       // wait until the web socket connection was opened
       await new Promise(process.nextTick);
       expect(fetchSpy.args[0][0]).toBe(
         'https://api.qminder.com/graphql/connection-key',
       );
-      expect(spy).toBeCalledWith(
+      expect(WebSocket).toBeCalledWith(
         `wss://api.qminder.com:443/graphql/subscription?rest-api-key=${keyValue}`,
       );
     });
   });
 
   describe('with websocket cleanup', () => {
-    afterEach(async () => {
```

Comment:
This block was testing the following line:

https://github.com/Qminder/javascript-api/pull/518/files#diff-b696092cf2ea982882e23e816d35f6a093ddf5f9bcee13768906934886fb41c8L304


I think it makes sense to detect connection hang-ups in a singular place and that will be ping-pongs and keepalives (at least for some time) so it made sense to not handle non-user-related hangup in the onclose event